pointcloud/basic_octree.go

```diff
@@ -129,3 +131,54 @@ func (octree *BasicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Ve
 func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
+
+// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
+func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, threshold int, buffer float64) (bool, error) {
+	if octree.MaxVal() < threshold {
+		return false, nil
+	}
+	switch octree.node.nodeType {
+	case internalNode:
+		ocbox, err := spatialmath.NewBox(
```

Comment:
[q] What is the logic behind using a box here and a sphere in the leafNodeFilled?

Reply:
An octree represents a cube divisible into octants, hence a box. While a radius around a given point is a sphere.

Reply:
Got it so it's more of a matter of boxes to figure out the intersection with cubic octants and the sphere for looking at point intersection? I can get behind that logic